pkg/network/tracer/tracer_linux_test.go

```diff
@@ -1715,3 +1717,93 @@ func TestBlockingReadCounts(t *testing.T) {
 
 	assert.Equal(t, uint64(n), conn.MonotonicSum().RecvBytes)
 }
+
+// GoTLS test
+
+// Test that we can capture HTTPS traffic from Go processes started after the
+// tracer.
+func TestHTTPGoTLSCaptureNewProcess(t *testing.T) {
+	const (
+		ServerAddr          = "localhost:8081"
+		ExpectedOccurrences = 10
+	)
+
+	if !httpSupported(t) {
+		t.Skip("HTTPS feature not available on pre 4.14.0 kernels")
+	}
+
+	if !httpsSupported(t) {
+		t.Skip("HTTPS feature not available supported for this setup")
+	}
+
+	// Setup
+	closeServer := httptest.HTTPServer(t, ServerAddr, httptest.Options{
+		EnableTLS: true,
+	})
+	defer closeServer()
+
+	cfg := config.New()
+	cfg.EnableHTTPMonitoring = true
+	cfg.EnableHTTPSMonitoring = true
+
+	tr, err := NewTracer(cfg)
+	require.NoError(t, err)
+	defer tr.Stop()
+	require.NoError(t, tr.RegisterClient("1"))
+
+	req, err := nethttp.NewRequest(nethttp.MethodGet, fmt.Sprintf("https://%s/%d/request", ServerAddr, nethttp.StatusOK), nil)
+	require.NoError(t, err)
+
+	clientBin := buildGoTLSClientBin(t)
```

Comment:
You can also use `go run` here instead.

Reply:
I am planning to use this binary in a future test for GoTLS support, so I think it makes more sense to build the binary once, and run it manually in each test. My understanding of `go run` is that it will rescan the sources and/or rebuild the binary each time it is called, which would be a bit slower that just building it once and reuse it.

Reply:
So should you be checking if the binary has already been built in `buildGoTLSClientBin`?

Reply:
Not necessarily. My plan is to use subtests when I'll add support for already running processes:

```go
func TestHTTPGoTLSAttachProbes(t *testing.T) {
	buildClientBin(t)

	t.Run("New Process", testHTTPGoTLSCaptureNewProcess)
	t.Run("Already running process", testHTTPGoTLSCaptureAlreadyRunning)
}
```